Classes/KIFTextInputTraitsOverrides.m

```diff
@@ -0,0 +1,75 @@
+//
+//  KIFTextInputTraitsOverrides.m
+//  KIF
+//
+//  Created by Harley Cooper on 1/31/18.
+//
+
+#import <objc/runtime.h>
+#import "KIFTextInputTraitsOverrides.h"
+
+static IMP autocorrectOriginalImp;
+static IMP smartDashesOriginalImp;
+static IMP smartQuotesOriginalImp;
+
+void KIFSetAutocorrect(BOOL setAutocorrectOn) {
+    if(!autocorrectOriginalImp) {
+        autocorrectOriginalImp = class_getMethodImplementation([UITextField class], @selector(autocorrectionType));
+    }
+
+    IMP autocorrectImp;
+    if(setAutocorrectOn) {
+        autocorrectImp = autocorrectOriginalImp;
+    } else if(!setAutocorrectOn) {
+        autocorrectImp = imp_implementationWithBlock(^(UITextField *_self) {
+            return UITextAutocorrectionTypeNo;
```

Comment:
Swizzling methods isn't a super cheap operation in the runtime. It'd save a little time (not really so bad, milliseconds per test) if you instead replaced each implementation once, and within the `implementationWithBlock` check the value of the static BOOL to return `TypeNo` if it is force disabled, otherwise returns `autocorrectOriginalImp(self, @selector(autocorrectionType))`.

In that case, this could become `KIFSwizzleTextFieldAutocorrect` (that gets called once at test loading time) to do the method replacement, and then expose a `KIFAllowAutocorrectDefaultBehavior(BOOL allowDefaultBehavior)` that just needs to set the static BOOL. The static BOOLs would default to NO, so you don't ever need to call `KIFAllowAutocorrectDefaultBehavior(NO)` in the framework (that's just how it would be by default on being swizzled).

Seems cheaper and easier to understand to me, but up to you.